### PR TITLE
interpreter: (riscv) add interpreter functions

### PR DESCRIPTION
**Pull request by superlopuh**
- [x] add tests for get_data

### CODE REVIEW

File: xdsl/interpreters/riscv.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        *,
+        bitwidth: int = 32,
+        data: dict[str, Any] | None = None,
+        custom_instructions: dict[str, CustomInstructionFn] | None = None,
+    ):
+        super().__init__()
+        self.module_op = module_op
+        self.bitwidth = bitwidth
+        if data is None:
+            data = RiscvFunctions.get_data(module_op)
```
Comment (PapyChacal) [inline xdsl/interpreters/riscv.py:76]: This is untested and the only used of `get_data`. Don't you want to test it if you use it?

Comment (PapyChacal) [reply xdsl/interpreters/riscv.py:76]: idem for `def pairs(...)` and the `__add__` helper, which AFAIU are meant to be used in `get_data`

**Review by PapyChacal — APPROVED** [review]
LGTM, just comments on what seems to be a significant part of the code untested? Not blocking for this though, mainly wondering!

File: xdsl/interpreters/riscv.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/interpreters/riscv.py:76]: Ah good point forgot to add tests for this

File: xdsl/interpreters/riscv.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+    data: list[_T]
+    offset: int = 0
+
+    def __add__(self, offset: int) -> Buffer[_T]:
+        """
+        Aliases the data list, so storing into the offset stores for all other references
+        to the list.
+        """
+        return Buffer(self.data, self.offset + offset // 4)
```
Comment (adutilleul) [inline xdsl/interpreters/riscv.py:48]: I don't mind too much but feels strange to have a generic data structure and a hardcoded "size" constant here.

**Review by adutilleul — APPROVED** [review]
LGTM!

**Pull request merged**